runtime/caml/eventring.h

```diff
@@ -0,0 +1,303 @@
+/**************************************************************************/
+/*                                                                        */
+/*                                 OCaml                                  */
+/*                                                                        */
+/*                          Sadiq Jaffer, Opsian                          */
+/*                 Stephen Dolan, University of Cambridge                 */
+/*                      Enguerrand Decorne, Tarides                       */
+/*                                                                        */
+/*   Copyright 2021 Opsian Ltd                                            */
+/*   Copyright 2020 University of Cambridge                               */
+/*   Copyright 2020 Tarides                                               */
+/*                                                                        */
+/*   All rights reserved.  This file is distributed under the terms of    */
+/*   the GNU Lesser General Public License version 2.1, with the          */
+/*   special exception on linking described in the file LICENSE.          */
+/*                                                                        */
+/**************************************************************************/
+
+#ifndef CAML_EVENTRING_H
+#define CAML_EVENTRING_H
+
+#include "mlvalues.h"
+#include <stdint.h>
+
+#ifdef CAML_INSTR
+#define CAML_EV_ALLOC(s) caml_ev_alloc(s)
+#define CAML_EV_ALLOC_FLUSH() caml_ev_alloc_flush()
+#else
+#define CAML_EV_ALLOC(s)      /**/
+#define CAML_EV_ALLOC_FLUSH() /**/
+#endif
```

Comment:
(This means code changes if we migrate things between the instrumented runtime and the normal runtime, but I think this is okay? Is there any expectation that users outside the compiler codebase would try to use these macros?)

Reply:
Yes, I think it would make sense to change those. The existing probes in the default runtime also don't need to be macros anymore. I left both as they in an attempt to make this PR easier to review. My intention was to follow up with a smaller PR that modifies those and also re-adds some of the instrumented probes that disappeared between 4.x and 5.0.